### ISSUE TITLE
Axl sqlquery response

### ISSUE THREAD

**Issue by crunchingnumbers** [corbob/PSCUCM#55]
This is not a bug so much as a lack of understanding on my part.

> In a sandbox environment I am getting correctly return rows with selected fields. 
> name              pkid
> ------              -----
> name value    pkid value

> In the environment I wish to use my code, I am getting each row returned as:
> row
> ----
> {name value 1, name value 2,  name value 3,....for each row}

I have tried debugging both environments and checking for missing cmdlets,  badly formatted requests by putting them in POSTMAN but have cannot figure out what I am missing. I feel sure that it is something pretty basic and obvious. 

Any pointers appreciated and thank you for the effort putting PSCUCM together and for your time

**Comment by crunchingnumbers** [comment]
I remembered I had figured this out - needed to think a little harder.
				if ($Entity -eq 'executeSQLQuery')
				 {
					$response = Invoke-WebRequest @IRMParams
					return select-xml -Content $response.Content -xPath '//row' | Select-Object -ExpandProperty node
				
				 }
				else
				{
					Invoke-WebRequest @IRMParams |
					Select-XML -XPath '//return' | 
					Select-Object -ExpandProperty Node
				}

**Comment by corbob** [comment]
IIRC there's actually an invoke command for a SQL query that I probably made for exactly this reason...

**Comment by crunchingnumbers** [comment]
Cory:



Thank you so much for looking at this. I had looked at it for 1-2 months on and off with the big assumption that it was something peculiar to the environment I was using. I debugged it and stepped through that code sequence so many times, I just became blind to it.  Finally this weekend I got a bit more granular and broke the instruction up in the console to see what was being piped to   select-object - then it clicked. I’m older and slower these days but thanks again.



Now working and a demo for my small team of Cisco engineers tomorrow to help them troubleshoot mainly SNR and Hunt Group errant users without the need to painstakingly use the CUCM admin interface.







Sincerely,



C. George Morri

•: ***@***.******@***.***>
